cylc/uiserver/app.py

```diff
@@ -385,7 +384,10 @@ def _list_ui_versions(path: Path) -> List[str]:
                 for version in path.glob('[0-9][0-9.]*')
                 if version
             ),
-            key=parse_version
+            key=lambda x: tuple(
+                int(x) if x.isdecimal() else x
+                for x in x.split('.')
+            )
```

Comment:
We might as well use the importlib-metadata interface for parsing the version here as it is already a dependency of cylc-flow.

Reply:
This is parsing a NodeJS semver version which is not guaranteed to be compatible with Python versioning syntax. I figure this custom sort key is good enough

Reply:
IMO, it's nicer to use the interface we've already got installed (and we know this package conforms to semver, because it's ours) but will leave it to you if you're keen not to.

Reply:
Have just realised that the `isdecimal` logic here does not work:

```python
>>> sorted(['1.1', '1.a1'], key=lambda x: tuple(int(x) if x.isdecimal() else x for x in x.split('.')))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: '<' not supported between instances of 'str' and 'int'
```

(it did at Python 2, but Python 3 doesn't allow comparisons of str and int)

Reply:
Ah woops, could just do?
```suggestion
            key=lambda x: tuple(
                int(i) if i.isdecimal() else -1
                for i in x.split('.')
            )
```

(Also realised I was confusingly redefining `x` in the comprehension)

Reply:
> (Also realised I was confusingly redefining x in the comprehension)

I spotted that, but it seems to work (confusingly), not quite sure how.

> Ah woops, could just do?

I would just use an existing version parser, there's a reason I've done this in the past!